sklearn/linear_model/tests/test_ransac.py

```diff
@@ -153,9 +153,14 @@ def test_ransac_resid_thresh_no_inliers():
     ransac_estimator = RANSACRegressor(base_estimator, min_samples=2,
                                        residual_threshold=0.0, random_state=0)
 
-    assert_raises_regexp(ValueError,
-                    "No inliers.*residual_threshold.*0\.0",
-                    ransac_estimator.fit, X, y)
+    msg = (
+        "RANSAC could not find valid consensus set, because"
+        " either the `residual_threshold` rejected all the samples or"
+        " `is_data_valid` and `is_model_valid` returned False for all"
```

Comment:
I believe you are referring to `is_data_valid` and `is_model_valid`. These are user-defined functions much like `residual_threshold` is a user-defined parameter.  When a user does not define these functions, RANSAC does not call them--they are initialized to `None` and evaluation is skipped. Unless you disagree, these may still be valid to refer to.

Perhaps a better message might read:
 
RANSAC could not find a valid consensus set. All `max_trials` iterations were skipped because each randomly chosen sub-sample either: didn't produce a valid set of inliers due to a small `residual_threshold` or was invalidated by `is_data_valid` or `is_model_valid` if either of these functions have been defined by the user.